docs/2.5. APIs - Query Parameters.md

```diff
@@ -12,60 +12,65 @@ Query APIs SHOULD support pagination of their API resources where the 'paged' tr
+To ensure that pagination does not result in resources being skipped, there SHOULD NOT be duplicate creation or update timestamps stored against resources of the same type (Node, Source, Flow etc). It is RECOMMENDED that these timestamps are stored with nanosecond resolution using a TAI timebase, which will allow clients to navigate collections based on a common understanding of time.
 
-* In order to ensure that pagination does not result in resources being skipped, it is important that there are no duplicate creation or update timestamps stored against resources of the same type (Node, Source, Flow etc). It is suggested (although not mandated) that these timestamps are stored with nanosecond resolution using a TAI timebase, which will allow clients to navigate collections based on a common understanding of time.
+The choice to page based on 'creation' or 'update' timestamp SHOULD depend on the client's intended use of the data:
 
-* The choice to page based on 'creation' or 'update' timestamp depends on the client's intended use of the data. Paging based on 'creation' time from the start of a collection provides the best mechanism to identify all resources held in a registry. Paging based on 'update' time from the end of a collection provides the best mechanism to watch for changing resources in the case that WebSocket subscriptions are not available.
+- Paging based on 'creation' time from the start of a collection provides the best mechanism to identify all resources held in a registry.
+- Paging based on 'update' time from the end of a collection provides the best mechanism to watch for changing resources in the case that WebSocket subscriptions are not available.
 
-* Paging through a collection for a second time, using the same cursors as previously should never result in new data appearing in the response payloads when using 'creation' ordering, provided the requested paging times are less than or equal to the maximum paging time held in the registry. However, note that as the registry is dynamic and resources may be updated or deleted a given cursor may return fewer results than it did previously. When using 'update' ordering and a query filter, new data may appear when cursors are used for a second time.
+Paging through a collection for a second time, using the same cursors as previously SHOULD NOT result in new data appearing in the response payloads when using 'creation' ordering, provided the requested paging times are less than or equal to the maximum paging time held in the registry. However, note that as the registry is dynamic and resources MAY be updated or deleted a given cursor MAY return fewer results than it did previously. When using 'update' ordering and a query filter, new data MAY appear when cursors are used for a second time.
```

Comment:
This is a little awkward because the sentence seems to begin from the point of view of the _client_, but the "SHOULD NOT" as a recommendation is addressing the _server_.

The instances of "may" in the second sentence were also really stating fact from the point of view of the client; as conformance language most of them apply to the server, so again, a bit awkward.

```suggestion
Paging through a collection for a second time, using the same cursors as previously SHOULD NOT result in new data appearing in the response payloads when using 'creation' ordering, provided the requested paging times are less than or equal to the maximum paging time held in the registry. However, note that as the registry is dynamic and resources could be updated or deleted a given cursor MAY return fewer results than it did previously. When using 'update' ordering and a query filter, new data MAY appear when cursors are used for a second time.
```